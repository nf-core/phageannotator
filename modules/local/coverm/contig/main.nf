process COVERM_CONTIG {
    tag "$meta.id"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/coverm:0.6.1--h1535e20_5':
        'biocontainers/coverm:0.6.1--h1535e20_5' }"

    input:
    tuple val(meta), path(bam_files)

    output:
    tuple val(meta), path("*_alignment_results.tsv")    , emit: alignment_results
    path "versions.yml"                                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    coverm \\
        contig \\
        --bam-files $bam_files \\
        --output-file ${prefix}_alignment_results.tsv \\
        --threads $task.cpus \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        coverm: \$(echo \$(coverm --version 2>&1) | sed 's/^.*coverm //; s/Using.*\$//' ))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_alignment_results.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        coverm: \$(echo \$(coverm --version 2>&1) | sed 's/^.*coverm //; s/Using.*\$//' ))
    END_VERSIONS
    """
}
