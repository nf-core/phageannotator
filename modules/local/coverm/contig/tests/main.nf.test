nextflow_process {

    name "Test Process: COVERM_CONTIG"
    script "../main.nf"
    process "COVERM_CONTIG"


    test("test1_paired_end.bam & test_2paired_end.bam") {

        when {
            process {
                """
                input[0] = [
                    [ id: 'all_samples' ],
                    [
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/bam/test1.bam', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/bam/test2.bam', checkIfExists: true)
                    ]
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() },
            )
        }
    }

    test("test1_paired_end.bam & test_2paired_end.bam - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id: 'all_samples' ],
                    [
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/bam/test1.bam', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/bam/test2.bam', checkIfExists: true)
                    ]
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() },
            )
        }
    }
}
