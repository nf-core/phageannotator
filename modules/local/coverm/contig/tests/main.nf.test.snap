{
    "test1_paired_end.bam & test_2paired_end.bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_alignment_results.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3f5dd63588c677b43b85a327268cd825"
                ],
                "alignment_results": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_alignment_results.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3f5dd63588c677b43b85a327268cd825"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-19T17:03:16.919022297"
    },
    "test1_paired_end.bam & test_2paired_end.bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_alignment_results.tsv:md5,f7624089f001fd3079bc43e3c3372a7e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3f5dd63588c677b43b85a327268cd825"
                ],
                "alignment_results": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_alignment_results.tsv:md5,f7624089f001fd3079bc43e3c3372a7e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3f5dd63588c677b43b85a327268cd825"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-19T17:03:08.329418215"
    }
}
